fl4health/clients/basic_client.py

```diff
@@ -248,6 +266,11 @@ def fit(self, parameters: NDArrays, config: Config) -> Tuple[NDArrays, int, Dict
         if not self.initialized:
             self.setup_client(config)
 
+            # If per_round_checkpointer not None and checkpoint exists load it and set attributes.
+            # Model not updated because FL restarted from most recent FL round (redo pre-empted round)
+            if self.per_round_checkpointer is not None and self.per_round_checkpointer.checkpoint_exists():
+                self.load_checkpoint()
```

Comment:
I'm curious to hear your thoughts, but I wonder if, rather than always loading from a checkpoint if it exists and a checkpointer has been defined, we should have a bool for loading from a state checkpoint? It's not perfect to add another constructor variable, but the situation I imagine is that you start a run with a `per_round_checkpointer` for preemption safety, but there is a pre-existing state file that you don't know about or thought would just be overridden, but this would actually load the state without the user explicitly requesting state loading? I might be over-thinking it, but figured I'd record the thought here.

Reply:
It definitely could be put in setup_client and maybe thats the direction we go, but I opted not too because I thought it made sense to have the load_checkpoint and save_checkpoint in the sample place (fit method).